src/test/java/com/artipie/jfr/ChunksAndSizeMetricsContentTest.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * The MIT License (MIT) Copyright (c) 2020-2021 artipie.com
+ * https://github.com/artipie/artipie/LICENSE.txt
+ */
+package com.artipie.jfr;
+
+import com.artipie.asto.Content;
+import com.artipie.asto.Key;
+import com.artipie.asto.Splitting;
+import com.artipie.asto.Storage;
+import com.artipie.asto.memory.InMemoryStorage;
+import io.reactivex.Flowable;
+import java.nio.ByteBuffer;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.core.Is;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests for ChunksAndSizeMetricsContent.
+ *
+ * @since 0.28.0
+ * @checkstyle ClassDataAbstractionCouplingCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ * @checkstyle ParameterNameCheck (500 lines)
+ * @checkstyle LocalFinalVariableNameCheck (500 lines)
+ * @checkstyle IllegalCatchCheck (500 lines)
+ */
+@SuppressWarnings(
+    {
+        "PMD.AvoidCatchingGenericException",
+        "PMD.AvoidThrowingRawExceptionTypes",
+        "PMD.EmptyCatchBlock",
+        "PMD.AvoidDuplicateLiterals"
+    }
+)
+class ChunksAndSizeMetricsContentTest {
+
+    /**
+     * Random one for all tests.
+     */
+    private static final Random RANDOM = new Random();
+
+    /**
+     * To check chunk count.
+     */
+    private AtomicInteger count;
+
+    /**
+     * To check size of content`s data.
+     */
+    private AtomicLong sum;
+
+    /**
+     * To consume data.
+     */
+    private Storage asto;
```

Comment:
hm, using storage in this test looks a bit of an overhead, I'd suggest using standard methods to process the publisher, for example:
```
Flowable.fromPublisher(content).lastOrError().blockingGet();
```
where content is the `ChunksAndSizeMetricsContent` instance